addons/mrp/models/mrp_production.py

```diff
@@ -662,16 +662,18 @@ def _onchange_producing(self):
     @api.onchange('lot_producing_id')
     def _onchange_lot_producing(self):
         if self.product_id.tracking == 'serial':
-            if self.env['stock.move.line'].search_count([
-                ('company_id', '=', self.company_id.id),
-                ('product_id', '=', self.product_id.id),
-                ('lot_id', '=', self.lot_producing_id.id),
-                ('state', '!=', 'cancel')
-            ]):
+            quants = self.env['stock.quant'].search([('product_id', '=', self.product_id.id),
```

Comment:
I understand we want to gather information about the quants rather than only the number of quants for the message, but we should see on a relatively large database if it does not slow down the flow too much.

Reply:
10-11 s to load on my local machine for 99969 (my loop logic wasn't clean enough for a round number...)  repeat serial numbers with a different location for each one. I think that's reasonable? The real slow down is in the dupe SN inventory adjustment :( The server times out with that many repeat serial numbers, but hopefully no one will ever let it get that bad?